File: src/app/npc.ts
import { Campaign } from "./campaign.interface";
import { Monster } from "./monster.interface";

export interface NPC {
    campaign: Campaign;
    id: number;
    name: string;
    race: string;
    age: number;
    class?: string;
    location?: string;
    description?: string;
    notes?: string;
    image?: string;
    hasMonsterStats?: Monster;
  }

File: backend/API/NPC.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;



namespace API
{
    public class NPC
    {

        public Campaign Campaign { get; set; }

        [Key]
        public int Id { get; set; }
        public string Name { get; set; }
        public string Race { get; set; }
        public int Age { get; set; }
        public string? Description { get; set; }

        public string? Notes { get; set; }


    }
}
